x-pack/solutions/observability/test/api_integration/apis/synthetics/add_monitor_project.ts

```diff
@@ -651,5 +651,212 @@ export default function ({ getService }: FtrProviderContext) {
         await security.role.delete(roleName);
       }
     });
+
+    it('project monitors - cannot update project monitors when user does not have access to a space in multi space use case', async () => {
+      const project = `test-project-${uuidv4()}`;
+      const username = 'admin';
+      const roleName = `synthetics_limited_space`;
+      const password = `${username}-password`;
+      const SPACE_ID_1 = `test-space-1-${uuidv4()}`;
+      const SPACE_ID_2 = `test-space-2-${uuidv4()}`;
+      const SPACE_NAME_1 = `test-space-name-1-${uuidv4()}`;
+      const SPACE_NAME_2 = `test-space-name-2-${uuidv4()}`;
+
+      await kibanaServer.spaces.create({ id: SPACE_ID_1, name: SPACE_NAME_1 });
+      await kibanaServer.spaces.create({ id: SPACE_ID_2, name: SPACE_NAME_2 });
+
+      try {
+        // Give user access to only SPACE_ID_1
+        await security.role.create(roleName, {
+          kibana: [
+            {
+              feature: {
+                uptime: ['all'],
+              },
+              spaces: [SPACE_ID_1],
+            },
+          ],
+        });
+        await security.user.create(username, {
+          password,
+          roles: [roleName],
+          full_name: 'a kibana user',
+        });
+
+        // Try to add a monitor to both spaces, but user only has access to one
+        const multiSpaceMonitor = {
+          ...projectMonitors.monitors[0],
+          spaces: [SPACE_ID_1, SPACE_ID_2],
+        };
+
+        const resp = await supertestWithoutAuth
+          .put(
+            `/s/${SPACE_ID_1}${SYNTHETICS_API_URLS.SYNTHETICS_MONITORS_PROJECT_UPDATE.replace(
+              '{projectName}',
+              project
+            )}`
+          )
+          .auth(username, password)
+          .set('kbn-xsrf', 'true')
+          .send({ monitors: [multiSpaceMonitor] });
+
+        expect([403, 404]).to.contain(resp.status);
+        expect([
+          `Kibana space does not exist, Error: Unauthorized to get ${SPACE_ID_2} space`,
+          'You do not have sufficient permissions to update monitors in all required spaces.',
+        ]).to.contain(resp.body.message);
```

Comment:
I don't understand exactly what are we testing here.. shouldn't the response always be the same?